components/omega/src/timeStepping/TimeStepper.cpp

```diff
@@ -0,0 +1,174 @@
+      return nullptr;
+   }
+
+   TimeStepper *NewTimeStepper;
+
+   switch (Type) {
+   case TimeStepperType::ForwardBackward:
+      NewTimeStepper =
+          new ForwardBackwardStepper(Name, Tend, AuxState, Mesh, MeshHalo);
+      break;
+   case TimeStepperType::RungeKutta4:
+      NewTimeStepper =
+          new RungeKutta4Stepper(Name, Tend, AuxState, Mesh, MeshHalo);
+      break;
+   case TimeStepperType::RungeKutta2:
+      NewTimeStepper =
+          new RungeKutta2Stepper(Name, Tend, AuxState, Mesh, MeshHalo);
+      break;
+   }
+
+   AllTimeSteppers.emplace(Name, NewTimeStepper);
+
+   return NewTimeStepper;
+}
+
+// Initialize the default time stepper
+int TimeStepper::init() {
+   int Err           = 0;
+   auto *DefMesh     = HorzMesh::getDefault();
+   auto *DefAuxState = AuxiliaryState::getDefault();
+   auto *DefHalo     = Halo::getDefault();
+   auto *DefTend     = Tendencies::getDefault();
+
+   TimeStepper::DefaultTimeStepper =
+       create("Default", TimeStepperType::ForwardBackward, DefTend, DefAuxState,
+              DefMesh, DefHalo);
+   return Err;
+}
+
+// Get the default time stepper
+TimeStepper *TimeStepper::getDefault() {
+   return TimeStepper::DefaultTimeStepper;
+}
+
+// Get time stepper by name
+TimeStepper *TimeStepper::get(const std::string &Name) {
+   // look for an instance of this name
+   auto it = AllTimeSteppers.find(Name);
+
+   // if found, return the pointer
+   if (it != AllTimeSteppers.end()) {
+      return it->second.get();
+
+      // otherwise print error and return null pointer
+   } else {
+      LOG_ERROR("TimeStepper::get: Attempt to retrieve non-existent "
+                "time stepper:");
+      LOG_ERROR("{} has not been defined or has been removed", Name);
+      return nullptr;
+   }
+}
+
+// Remove time stepper by name
+void TimeStepper::erase(const std::string &Name) {
+   AllTimeSteppers.erase(Name);
+}
+
+// Remove all time steppers
+void TimeStepper::clear() { AllTimeSteppers.clear(); }
+
+// Get time stepper name
+std::string TimeStepper::getName() const { return Name; }
+
+// Get time stepper type
+TimeStepperType TimeStepper::getType() const { return Type; }
+
+// LayerThickness1(TimeLevel1) = LayerThickness(TimeLevel2) + Coeff *
+// LayerThicknessTend
+void TimeStepper::updateThicknessByTend(OceanState *State1, int TimeLevel1,
+                                        OceanState *State2, int TimeLevel2,
+                                        Real Coeff) const {
+
+   const auto &LayerThick1    = State1->LayerThickness[TimeLevel1];
+   const auto &LayerThick2    = State2->LayerThickness[TimeLevel2];
+   const auto &LayerThickTend = Tend->LayerThicknessTend;
+   const int NVertLevels      = LayerThickTend.extent_int(1);
+
+   parallelFor(
+       "updateThickByTend", {Mesh->NCellsAll, NVertLevels},
+       KOKKOS_LAMBDA(int ICell, int K) {
+          LayerThick1(ICell, K) =
+              LayerThick2(ICell, K) + Coeff * LayerThickTend(ICell, K);
+       });
```

Comment:
Do we want to do the `KChunk` looping here?

Reply:
I think this code is so simple that auto-vectorization should work. We can always add `KChunk` looping later if profiling shows otherwise.

Reply:
Sounds good to me, thanks.